ceno_zkvm/src/e2e.rs

```diff
@@ -546,10 +588,13 @@ pub fn run_e2e_with_checkpoint<
     let exit_code = emul_result.exit_code;
 
     if let Checkpoint::PrepWitnessGen = checkpoint {
-        return (
-            (None, None),
-            Box::new(move || _ = generate_witness(&system_config, emul_result, &program, false)),
```

Comment:
why previously this didn't respect the `is_mock_proving ` flag? is that intend?

Reply:
yes it is on purpose, but I feel the naming is confused. 

**what's `is_mock_proving` do**
this is like halo2 mock_prover functionality, which is checking satisfying of al constrains on traces directly without generate proof.

In here, when we run e2e and halt **before** `generate_witness`, this imply we are going to benchmark generate_witness performance. So we skip mock proving check on `generate_witness`

But I feel we should have its control via "sanity_check" feature, wdyt? (we can do it in separate PR)